src/components/molecules/PollBox/PollBox.tsx

```diff
@@ -0,0 +1,115 @@
+import React, { useCallback, useMemo } from "react";
+import { useForm, useFieldArray } from "react-hook-form";
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import { faPaperPlane, faPlus } from "@fortawesome/free-solid-svg-icons";
+
+import { MAX_POLL_CHOICES } from "settings";
+
+import { getVenueOwners } from "api/admin";
+
+import { PollValues, PollQuestion } from "types/chat";
+
+import { InputField } from "components/atoms/InputField";
+
+import { useVenueId } from "hooks/useVenueId";
+import { useRoles } from "hooks/useRoles";
+import { useUser } from "hooks/useUser";
+
+import "./PollBox.scss";
+
+export interface PollBoxProps {
+  createPoll: (props: PollValues) => void;
+}
+
+const DEFAULT_QUESTION: Partial<PollQuestion> = { name: "" };
+const DEFAULT_VALUES = {
+  topic: "",
+  questions: [DEFAULT_QUESTION, DEFAULT_QUESTION],
+};
+
+export const PollBox: React.FC<PollBoxProps> = ({ createPoll }) => {
+  const venueId = useVenueId();
+  const { userRoles } = useRoles();
+  const { user } = useUser();
+
+  const userId = user?.uid;
+  const isAdmin = Boolean(userRoles?.includes("admin"));
+
+  const { register, control, handleSubmit, reset, watch } = useForm<PollValues>(
+    {
+      defaultValues: DEFAULT_VALUES,
+    }
+  );
+  const { fields, append } = useFieldArray({ name: "questions", control });
+  const [question1, question2] = watch("questions");
+  const topic = watch("topic");
+
+  const onPollSubmit = handleSubmit(async ({ topic, questions }) => {
+    if (!venueId || !userId) return;
+
+    const owners = await getVenueOwners(venueId);
+
+    const poll = {
+      canBeDeleted: isAdmin && owners.includes(userId),
+      topic,
+      questions: questions.map(({ name }, id) => ({ name, id })),
+    };
+    createPoll(poll);
+    reset();
+  });
+
+  const isDisabled = !(topic && question1.name && question2.name);
+
+  const addChoice = useCallback(() => append(DEFAULT_QUESTION), [append]);
+  const showAppend = useCallback(
+    (index) => index + 1 === fields.length && MAX_POLL_CHOICES > fields.length,
```

Comment:
pass the question number, not index.

Rename to `checkIfChoiceCanBeAppended`

Reply:
fixed in f2fa24aadcae7547799e463bd593e8a7a2cf05b5